### ISSUE TITLE
Add current domain condition plugin

### ISSUE THREAD

**Issue by artreaktor** [agentrickard/domain#190]

**Comment by agentrickard** [comment]
How is this different from the ContextProvider plugin?

Drupal\domain\ContextProvider\CurrentDomainContext

**Comment by agentrickard** [comment]
Apparently I am not alone in my confusion. Thanks @chx.

https://www.drupal.org/node/2344045

**Comment by artreaktor** [comment]
Good question :)
Not sure I'm understanding the difference between ContextProvider and Condition plugin. But current domain context is not available in page_manager/panels, so I decided we need another plugin.
However, something wrong here, as I have both options on the block settings page now:
![domain_context5](https://cloud.githubusercontent.com/assets/1705475/13055782/f39d5288-d419-11e5-9aba-a62e1001a9f6.JPG)
Here is how my page_manager/panel configuration workflow is looking now:
![domain_context4](https://cloud.githubusercontent.com/assets/1705475/13055951/e63a360a-d41a-11e5-8ec3-e649c245a437.JPG)

1. Add domain context:
![domain_context](https://cloud.githubusercontent.com/assets/1705475/13056096/cd107c88-d41b-11e5-8f57-4ff6bdc6a80b.JPG)
(why should we explicitly define an entity here? it relates to each entity, not only domain. core bug?)

2. Add selection condition
![domain_context2](https://cloud.githubusercontent.com/assets/1705475/13055976/0ce4e138-d41b-11e5-931c-fa059e6e0063.JPG)
The "standard" Domain condition doesn't fit my needs, as it checks whether the domain is enabled (active), so can't be used to filter-out domain-related content.

I didn't find any other working solution, so added new condition plugin. 
Obviously we need change something, so I opened to your suggestions

**Comment by agentrickard** [comment]
I think that the Condition Plugin extends and makes generic the ContextProvider. 

Perhaps remove the form elements from the ContextProvider and only provide them in the Condition?

Note that we have a test -- Drupal\domain\Tests\Condition\DomainConditionTest -- so that should help,

The current Context provider does check active domain, not status, it may be that Page Manager is generating that based on the entity definition.

**Comment by agentrickard** [comment]
The best explanation is https://www.drupal.org/node/2344045#comment-9265459 and if you can find EclipseGC on IRC, I'm sure he can help.

**Comment by EclipseGc** [comment]
Conditions (and other plugin types) consume contexts. A context is an object required at run time by a plugin. In the case of conditions, we can look at the node type condition. I cannot very well tell you what type of node you are on if you have no node object to evaluate, therefore, the node type condition requires a node to operate. This is a context.

Context providers, in core, provide a context to the block system. Blocks in turn leverage conditions in order to determine visibility. Both plugin types can use context(s). This is doubly confusing because core doesn't actually limit available conditions by context (blocks luckily do). This issue would solve that: https://www.drupal.org/node/2632434

Hope this helps.

**Comment by artreaktor** [comment]
Ken, Kris, thanks for explanation! It takes some time for me to digest this due to my 2 weeks D8 experience :)

Ken, 
> The current Context provider does check active domain, not status

means I'm trying to implement what is already done, but it not worked for me in panels:
It seems that it unable to get current domain with 
`$domain = $this->getContextValue('domain');`
and gets what is stored in condition settings instead and always evaluates to TRUE
Not sure it is correct workaround, but this is working for me:
[https://dl.dropboxusercontent.com/u/42690200/CurrentDomain.zip](url)

**Comment by agentrickard** [comment]
That suggests that Panels is behaving differently than the block system?

**Comment by artreaktor** [comment]
As I understood the Block system uses "generic" ContextProvider, and Panels uses Domain condition plugin, which probably contains bug

**Comment by agentrickard** [comment]
There are three conditions in play here that we need to be aware of:

* $domains = $this->configuration['domains'];
** Lists the domains set in the configuration form of the Selection Condition.
* $domain = $this->getContextValue('domain');
** Lists the Static Context set by Panels. 
* $active_domain = \Drupal::service('domain.negotiator')->getActiveDomain();
** Lists the active domain

It looks like Panels treatment of getContextValue() is different than the block system's.

**Comment by agentrickard** [comment]
Try this patch.

```
diff --git a/domain/src/Plugin/Condition/Domain.php b/domain/src/Plugin/Condition/Domain.php
index 5e6a489..351b519 100644
--- a/domain/src/Plugin/Condition/Domain.php
+++ b/domain/src/Plugin/Condition/Domain.php
@@ -89,7 +89,10 @@ class Domain extends ConditionPluginBase {
     if (empty($domains) && !$this->isNegated()) {
       return TRUE;
     }
-    $domain = $this->getContextValue('domain');
+    if ($domain = $this->getContextValue('domain')) {
+      return TRUE;
+    }
+    $domain = \Drupal::service('domain.negotiator')->getActiveDomain();
     // NOTE: The block system handles negation for us.
     return (bool) in_array($domain->id(), $domains);
   }
```

**Comment by agentrickard** [comment]
I have to say, I can't make Panels work at all without adding a Static Context, which seems like a Panels bug.

**Comment by artreaktor** [comment]
See #193

**Comment by agentrickard** [comment]
Fixed in #197
